test/src/kernels/PMassKernel.C

```diff
@@ -0,0 +1,41 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+#include "PMassKernel.h"
+
+template <>
+InputParameters
+validParams<PMassKernel>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addRangeCheckedParam<Real>("p", 2.0, "p>=1.0", "The exponent p");
+  return params;
+}
+
+PMassKernel::PMassKernel(const InputParameters & parameters)
+  : Kernel(parameters), _p(getParam<Real>("p") - 2.0)
```

Comment:
The parameter description string says that `_p` is an exponent, but you are subtracting 2 from it here. Perhaps the description is not complete?